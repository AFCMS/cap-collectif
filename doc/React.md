### How to write localizable components

Just import the appropriate [component](https://github.com/yahoo/react-intl/wiki#the-react-intl-module) from `react-intl`

- For localizable text use
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage).

- For date and time:
[`<FormattedDate>`](https://github.com/yahoo/react-intl/wiki/Components#formatteddate)
[`<FormattedTime>`](https://github.com/yahoo/react-intl/wiki/Components#formattedtime)
[`<FormattedRelative>`](https://github.com/yahoo/react-intl/wiki/Components#formattedrelative)

- For numbers and currencies:
[`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber)
[`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural)

- If possible, do not use `<FormattedHTMLMessage>`, see how to use *Rich Text Formatting* with
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage)

- When you need an imperative formatting API, use the [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injectintl) High-Order Component.
