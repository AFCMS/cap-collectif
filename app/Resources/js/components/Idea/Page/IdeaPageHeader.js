import React from 'react';
import classNames from 'classnames';
import UserAvatar from '../../User/UserAvatar';
import IdeaPageHeaderInfos from './IdeaPageHeaderInfos';

const IdeaPageHeader = React.createClass({
  propTypes: {
    idea: React.PropTypes.object.isRequired,
    className: React.PropTypes.string,
  },

  getDefaultProps() {
    return {
      className: '',
    };
  },

  render() {
    const { idea, className } = this.props;

    const classes = {
      idea__header: true,
    };
    classes[className] = true;

    return (
      <div className={classNames(classes)}>
        <h1 className="h1" id="idea-title">
          {idea.title}
        </h1>
        <div className="media">
          <UserAvatar className="pull-left" user={idea.author} />
          <div className="media-body">
            <IdeaPageHeaderInfos idea={idea} />
          </div>
        </div>
      </div>
    );
  },
});

export default IdeaPageHeader;
