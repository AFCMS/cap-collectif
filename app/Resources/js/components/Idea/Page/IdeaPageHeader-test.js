/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import IdeaPageHeader from './IdeaPageHeader';

const props = {
  idea: {
    id: 1,
    title: 'Title',
  },
};

describe('<IdeaPageHeader />', () => {
  it('it should render a title, a user avatar and idea infos', () => {
    const wrapper = shallow(<IdeaPageHeader {...props} />);
    expect(wrapper.find('div.idea__header')).toHaveLength(1);
    expect(wrapper.find('#idea-title')).toHaveLength(1);
    const title = wrapper.find('h1');
    expect(title).toHaveLength(1);
    expect(title.text()).toEqual(props.idea.title);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render a div with provided class name', () => {
    const wrapper = shallow(<IdeaPageHeader {...props} className="css-class" />);
    expect(wrapper.find('div.idea__header.css-class')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
