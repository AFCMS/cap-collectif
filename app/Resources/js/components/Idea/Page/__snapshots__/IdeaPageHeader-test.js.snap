// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdeaPageHeader /> it should render a div with provided class name 1`] = `
<div
  className="idea__header css-class"
>
  <h1
    className="h1"
    id="idea-title"
  >
    Title
  </h1>
  <div
    className="media"
  >
    <UserAvatar
      anchor={true}
      className="pull-left"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      size={45}
      style={Object {}}
      user={null}
    />
    <div
      className="media-body"
    >
      <Connect(IdeaPageHeaderInfos)
        idea={
          Object {
            "id": 1,
            "title": "Title",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<IdeaPageHeader /> it should render a title, a user avatar and idea infos 1`] = `
<div
  className="idea__header "
>
  <h1
    className="h1"
    id="idea-title"
  >
    Title
  </h1>
  <div
    className="media"
  >
    <UserAvatar
      anchor={true}
      className="pull-left"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      size={45}
      style={Object {}}
      user={null}
    />
    <div
      className="media-body"
    >
      <Connect(IdeaPageHeaderInfos)
        idea={
          Object {
            "id": 1,
            "title": "Title",
          }
        }
      />
    </div>
  </div>
</div>
`;
